python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -113,14 +131,54 @@ def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
     def add_kwargs_from_args(
         cls, args: argparse.Namespace, kwargs: typing.Dict[str, typing.Any]
     ) -> None:
-        """Process custom --hostname, --port and --unit arguments."""
-        cls.hostname = (
-            f"m1m3cam-aircomp{kwargs['index']:02d}.cp.lsst.org"
-            if args.hostname is None
-            else args.hostname
-        )
+        """Process custom --grace-period, --host, --port and --unit
+        arguments."""
+        cls.grace_period = args.grace_period
+        cls.host = args.host
         cls.port = args.port
-        cls.unit = kwargs["index"] if args.unit is None else args.unit
+        cls.unit = args.unit
+
+    async def configure(self, config):
+        self.config = config
+        ci = list(
+            filter(
+                lambda i: i["sal_index"] == self.salinfo.index, self.config.instances
+            )
+        )
+        if len(ci) == 0:
+            raise RuntimeError(
+                f"Cannot find configuration for index {self.salinfo.index},"
+                "at least sal_index entry must be provided"
+            )
+        elif len(ci) > 1:
+            raise RuntimeError(
+                f"Multiple configuration instances matches index {self.salinfo.index},"
+                "please check configuration file"
+            )
+        else:
+            ci = ci[0]
+        if self.grace_period is None:
+            self.grace_period = ci.get("grace_period")
+            if self.grace_period is None:
+                self.grace_period = 3600
+        if self.host is None:
+            self.host = ci.get("hosta")
+            if self.host is None:
+                self.host = f"m1m3cam-aircomp{self.salinfo.index:02d}.cp.lsst.org"
+
+        if self.port is None:
+            self.port = ci.get("port")
```

Comment:
Same as above, consider doing 

```
self.port = ci.get("port", 502)
```

Also, consider creating a constant with the default value, instead of hard coding it here..